taskcat/_cfn/threaded.py

```diff
@@ -228,3 +230,55 @@ def _group_stacks(stacks: Stacks) -> List[dict]:
                 stacks_by_client[client] = {"Client": client, "Stacks": []}
             stacks_by_client[client]["Stacks"].append(stack)
         return [stacks_by_client[r] for r in stacks_by_client]
+
+    @staticmethod
+    def list_stacks(profiles, regions):
+        stacks = fan_out(
+            Stacker._list_per_profile,
+            {"regions": regions, "boto_cache": Boto3Cache()},
+            profiles,
+            threads=8,
+        )
+        return [stack for sublist in stacks for stack in sublist]
+
+    @staticmethod
+    def _list_per_profile(profile, regions, boto_cache):
+        stacks = fan_out(
+            Stacker._get_taskcat_stacks,
+            {"boto_cache": boto_cache, "profile": profile},
+            regions,
+            threads=len(regions),
+        )
+        return [stack for sublist in stacks for stack in sublist]
+
+    @staticmethod
+    def _get_taskcat_stacks(region, boto_cache: Boto3Cache, profile: str):
+        stacks = []
+        try:
+            cfn = boto_cache.client("cloudformation", profile=profile, region=region)
+            for page in cfn.get_paginator("describe_stacks").paginate():
+                for stack_props in page["Stacks"]:
+                    if stack_props.get("ParentId"):
+                        continue
+                    stack_id = stack_props["StackId"]
+                    stack_name = stack_id.split("/")[1]
+                    stack = {
+                        "region": region,
+                        "profile": profile,
+                        "stack-id": stack_id,
+                        "stack-name": stack_name,
+                    }
+                    for tag in stack_props["Tags"]:
+                        k, v = (tag["Key"], tag["Value"])
+                        if k.startswith("taskcat-"):
+                            stack[k] = v
+                    if stack.get("taskcat-id"):
```

Comment:
I'm having trouble understanding why you would convert it into a UUID.

Reply:
just because everything else wants the id as a uuid, but probably not needed here